molgenis-metadata-manager/src/main/java/org/molgenis/metadata/manager/mapper/AttributeMapper.java

```diff
@@ -0,0 +1,222 @@
+	{
+		editorAttributes.forEach(editorAttribute ->
+		{
+			EditorAttributeIdentifier editorAttributeParent = editorAttribute.getParent();
+			if (editorAttributeParent != null)
+			{
+				String attributeId = editorAttribute.getId();
+				String attributeParentId = editorAttributeParent.getId();
+				Attribute attribute = attributeMap.get(attributeId);
+				Attribute parentAttribute = attributeMap.get(attributeParentId);
+				attribute.setParent(parentAttribute);
+			}
+		});
+		return attributeMap.values();
+	}
+
+	ImmutableList<EditorAttribute> toEditorAttributes(Iterable<Attribute> attributes)
+	{
+		return ImmutableList.copyOf(stream(attributes.spliterator(), false).map(this::toEditorAttribute).iterator());
+	}
+
+	private EditorAttribute toEditorAttribute(Attribute attribute)
+	{
+		String id = attribute.getIdentifier();
+		String name = attribute.getName();
+		String type = attribute.getDataType() != null ? attribute.getDataType().toString() : null;
+		EditorAttributeIdentifier parent = attributeReferenceMapper.toEditorAttributeIdentifier(attribute.getParent());
+		EditorEntityTypeIdentifier refEntityType = entityTypeReferenceMapper
+				.toEditorEntityTypeIdentifier(attribute.getRefEntity());
+		EditorAttributeIdentifier mappedByEntityType = attributeReferenceMapper
+				.toEditorAttributeIdentifier(attribute.getMappedBy());
+		EditorSort orderBy = sortMapper.toEditorSort(attribute.getOrderBy());
+		String expression = attribute.getExpression();
+		boolean nullable = attribute.isNillable();
+		boolean auto = attribute.isAuto();
+		boolean visible = attribute.isVisible();
+		String label = attribute.getLabel();
+		ImmutableMap<String, String> i18nLabel = toI18nLabel(attribute);
+		String description = attribute.getDescription();
+		ImmutableMap<String, String> i18nDescription = toI18nDescription(attribute);
+		boolean aggregatable = attribute.isAggregatable();
+		ImmutableList<String> enumOptions = ImmutableList.copyOf(attribute.getEnumOptions());
+		Long rangeMin = attribute.getRangeMin();
+		Long rangeMax = attribute.getRangeMax();
+		boolean readonly = attribute.isReadOnly();
+		boolean unique = attribute.isUnique();
+		ImmutableList<EditorTagIdentifier> tags = tagMapper.toEditorTags(attribute.getTags());
+		String visibleExpression = attribute.getVisibleExpression();
+		String validationExpression = attribute.getValidationExpression();
+		String defaultValue = attribute.getDefaultValue();
+		Integer sequenceNumber = attribute.getSequenceNumber() != null ? attribute.getSequenceNumber() : 0;
+		return EditorAttribute
+				.create(id, name, type, parent, refEntityType, mappedByEntityType, orderBy, expression, nullable, auto,
+						visible, label, i18nLabel, description, i18nDescription, aggregatable, enumOptions, rangeMin,
+						rangeMax, readonly, unique, tags, visibleExpression, validationExpression, defaultValue,
+						sequenceNumber);
+	}
+
+	private ImmutableMap<String, String> toI18nLabel(Attribute attribute)
```

Comment:
i18n is part of the metadata of EntityTypes, so it should be here, but I don't really have an idea on how to use it though (also TODO/ out of scope in mockups)